desktop/apps/article/components/App.js

```diff
@@ -44,14 +45,33 @@ export default class App extends React.Component {
     const { article } = this.props
 
     return (
-      <div>
+      <Fragment>
+        <EditPortal article={article} />
+        {this.getArticleLayout()}
+      </Fragment>
+    )
+  }
+}
+
+class EditPortal extends React.Component {
+  static propTypes = {
+    article: PropTypes.object
+  }
+
+  render () {
+    const { article } = this.props
+
+    // Only render on client-side
+    if (typeof window !== 'undefined') {
```

Comment:
Wasn't too crazy about this, but we don't have access to `document` on server side. Any cooler ideas?

Reply:
Oh no, that's literally the textbook way to do it and you see it all over SSR-related apps / libs 😄

Reply:
@damassi I like this: 

```
  render () {
    const { article } = this.props

    try {
      return ReactDOM.createPortal(
        <EditButton
          channelId={article.channel_id}
          slug={article.slug}
        />,
        document.getElementById('react-portal')
      )
    } catch (e) {
      return false
    }
  }
```

Because it also works nicely with tests when jsdom has `window` but can't find the element.  ie, the currently failing tests: `Invariant Violation: Target container is not a DOM element.`

Reply:
Oh that's interesting. I don't have the answer off the top of my head  but I would maybe check for window in the render method and if `undefined` return null. Maybe that fixes it?

Reply:
Mm that's what I have currently? The problem is that JSDOM does have a window so it tries to render but fails on `document.getElementById('react-portal')`. I'm digging this try/catch way so I'll stick with that :)